components/automate-cli/pkg/verifyserver/models/batchcheck.go

```diff
@@ -78,6 +81,191 @@ type Config struct {
+			c.ExternalPG.PGInstanceURL = awsManagedServicesConfig.ManagedRdsInstanceURL
+
+			// pg root-ca might be nil in pre deploy state
+			c.ExternalPG.PGRootCert = awsManagedServicesConfig.ManagedRdsCertificate
+			c.ExternalPG.PGSuperuserName = awsManagedServicesConfig.ManagedRdsSuperuserUsername
+			c.ExternalPG.PGSuperuserPassword = awsManagedServicesConfig.ManagedRdsSuperuserPassword
+
+			c.ExternalOS.OSDomainName = awsManagedServicesConfig.ManagedOpensearchDomainName
+			c.ExternalOS.OSDomainURL = awsManagedServicesConfig.ManagedOpensearchDomainURL
+
+			// os root-ca might be nil in pre deploy state
+			c.ExternalOS.OSCert = awsManagedServicesConfig.ManagedOpensearchCertificate
+			c.ExternalOS.OSUserPassword = awsManagedServicesConfig.ManagedOpensearchUserPassword
+			c.ExternalOS.OSUsername = awsManagedServicesConfig.ManagedOpensearchUsername
+			c.ExternalOS.OSRoleArn = awsManagedServicesConfig.AwsOsSnapshotRoleArn
+		}
+	}
+
+	if haConfig.IsExistingInfra() {
+		existingInfraConfig := haConfig.ExistingInfra.Config
+		c.Hardware.AutomateNodeIps = existingInfraConfig.AutomatePrivateIps
+		c.Hardware.ChefInfraServerNodeIps = existingInfraConfig.ChefServerPrivateIps
+		automateConfig := haConfig.Automate.Config
+		if automateConfig.EnableCustomCerts {
+			c.appendCertsByIpToNodeCerts(automateConfig.CertsByIP, haConfig.ExistingInfra.Config.AutomatePrivateIps, automateConfig.PrivateKey, automateConfig.PublicKey, "", "")
+		}
+
+		chefserverConfig := haConfig.ChefServer.Config
+		if chefserverConfig.EnableCustomCerts {
+			c.appendCertsByIpToNodeCerts(chefserverConfig.CertsByIP, haConfig.ExistingInfra.Config.ChefServerPrivateIps, chefserverConfig.PrivateKey, chefserverConfig.PublicKey, "", "")
+		}
+
+		if !haConfig.IsExternalDb() {
+			c.Hardware.PostgresqlNodeIps = existingInfraConfig.PostgresqlPrivateIps
+			c.Hardware.OpenSearchNodeIps = existingInfraConfig.OpensearchPrivateIps
+			postgresqlConfig := haConfig.Postgresql.Config
+			if postgresqlConfig.EnableCustomCerts {
+				c.appendCertsByIpToNodeCerts(postgresqlConfig.CertsByIP, haConfig.ExistingInfra.Config.PostgresqlPrivateIps, postgresqlConfig.PrivateKey, postgresqlConfig.PublicKey, "", "")
+			}
+
+			openSearchConfig := haConfig.Opensearch.Config
+			if openSearchConfig.EnableCustomCerts {
+				c.appendCertsByIpToNodeCerts(openSearchConfig.CertsByIP, haConfig.ExistingInfra.Config.OpensearchPrivateIps, openSearchConfig.PrivateKey, openSearchConfig.PublicKey, openSearchConfig.AdminKey, openSearchConfig.AdminCert)
+			}
+		}
+
+		if haConfig.IsExternalDb() {
+
+			externalPgConfig := haConfig.External.Database.PostgreSQL
+			c.ExternalPG.PGDbUserName = externalPgConfig.DbuserUsername
+			c.ExternalPG.PGDbUserPassword = externalPgConfig.DbuserPassword
+			c.ExternalPG.PGInstanceURL = externalPgConfig.InstanceURL
+
+			// pg root-ca might be nil in pre deploy state
+			c.ExternalPG.PGRootCert = externalPgConfig.PostgresqlRootCert
```

Comment:
is there a story for taking pgrootca as input from config.toml

Reply:
could you please pin the stories that will later on populate any field, in the PR description

Reply:
There is no story, currently.